Formula/swift.rb

```diff
@@ -23,56 +25,170 @@ class Swift < Formula
   end
 
   resource "indexstore-db" do
-    url "https://github.com/apple/indexstore-db/archive/swift-5.2.5-RELEASE.tar.gz"
-    sha256 "cefe69f9b63869acee0564d38d8eb98f449647db9c8df7cd1c59538f506e7f1e"
+    url "https://github.com/apple/indexstore-db/archive/swift-5.4.2-RELEASE.tar.gz"
+    sha256 "876f170ecbce1461cc21509a52d11b4e79a045f6348e0d8f1c643e9e6e0e1624"
   end
 
   resource "sourcekit-lsp" do
-    url "https://github.com/apple/sourcekit-lsp/archive/swift-5.2.5-RELEASE.tar.gz"
-    sha256 "2cb2dffc585a068cefa3bf33d873394f3bccddf8e5e7269889d6960f387ddcfc"
+    url "https://github.com/apple/sourcekit-lsp/archive/swift-5.4.2-RELEASE.tar.gz"
+    sha256 "2eff815309fa34bcb18a70298e16deb974862806a449c93eb245162030fe4d73"
+  end
+
+  resource "swift-driver" do
+    url "https://github.com/apple/swift-driver/archive/swift-5.4.2-RELEASE.tar.gz"
+    sha256 "9907e6d41236cf543a43a89b5ff67b6cb12474692f96069908d4b6f92b617518"
   end
 
+  resource "swift-tools-support-core" do
+    url "https://github.com/apple/swift-tools-support-core/archive/swift-5.4.2-RELEASE.tar.gz"
+    sha256 "a4bc991cf601fe0f45edc7d0a6248f1a19def4d149b3e86b37361f34b0ecbd2c"
+  end
+
+  # To find the version to use, check the release/#{version.major_minor} entry of:
+  # https://github.com/apple/swift/blob/swift-#{version}-RELEASE/utils/update_checkout/update-checkout-config.json
+  resource "swift-argument-parser" do
+    url "https://github.com/apple/swift-argument-parser/archive/0.4.1.tar.gz"
+    sha256 "6743338612be50a5a32127df0a3dd1c34e695f5071b1213f128e6e2b27c4364a"
+  end
+
+  # Similar scenario as above - see the above file to find the version to use here.
+  resource "yams" do
+    url "https://github.com/jpsim/Yams/archive/4.0.2.tar.gz"
+    sha256 "8bbb28ef994f60afe54668093d652e4d40831c79885fa92b1c2cd0e17e26735a"
+  end
+
+  # Don't build and install some test binaries.
+  # Remove with Swift 5.5.
+  patch do
+    url "https://github.com/apple/swift/commit/86c40574f594f4f7b4b25bb02cc2389e1328c200.patch?full_index=1"
+    sha256 "6d22d159b8c16ec14c4ec1e7ea258ffa489f932842b0f4be930d0f9660cbe9b4"
+  end
+
+  # Fix arm64 build not being able to use the arm64e standard library.
+  # Based on https://github.com/apple/swift/pull/39083.
+  # This patch has been adapted for Swift 5.4
+  # and should be updated with Swift 5.5 to more closely match the upstream PR.
+  patch :DATA
+
   def install
     workspace = buildpath.parent
     build = workspace/"build"
 
-    toolchain_prefix = "/Swift-#{version}.xctoolchain"
-    install_prefix = "#{toolchain_prefix}/usr"
+    toolchain_prefix = ""
+    install_prefix = "/libexec"
+    on_macos do
+      toolchain_prefix = "/Swift-#{version}.xctoolchain"
+      install_prefix = "#{toolchain_prefix}/usr"
+    end
 
     ln_sf buildpath, workspace/"swift"
-    resources.each { |r| r.stage(workspace/r.name) }
+    resources.each do |r|
+      next if r.name == "six"
+
+      r.stage(workspace/r.name)
+    end
```

Comment:
Since we don't have `six` as a resource we can revert this part back (could be done later to not to retrigger the build)

```suggestion
    resources.each { |r| r.stage(workspace/r.name) }
```

Reply:
I could probably cheat and just amend the commit while merging via `pr-pull`.

Reply:
Bottles are 3 GB. `pr-pull` might take a while.

Reply:
I've dealt with LLVM back when it was all statically linked and we didn't have `pr-publish`, it'll probably be fine.